aten/src/ATen/TensorIterator.cpp

```diff
@@ -1281,12 +1302,25 @@ void TensorIteratorBase::build(TensorIteratorConfig& config) {
   view_offsets_ = DimVector(ndim_offsets, 0);
 }
 
-void TensorIterator::set_output(int64_t output_idx, IntArrayRef sizes, IntArrayRef strides, TensorOptions options, DimnameList names) {
+void TensorIterator::set_output(
+    int64_t output_idx, IntArrayRef sizes, IntArrayRef strides,
+    TensorOptions options, DimnameList names,
+    c10::optional<TensorQuantizationOptions> quant_options) {
   auto& op = operands_[output_idx];
   TORCH_INTERNAL_ASSERT_DEBUG_ONLY(output_idx < num_outputs_);
   if (!op.tensor.defined()) {
       if (strides.empty()) {
-          op.tensor = at::empty(sizes, options);
+          if (options.dtype() == at::kQInt8 || options.dtype() == at::kQUInt8 ||
+              options.dtype() == at::kQInt32) {
+            // quantized path
+            TORCH_INTERNAL_ASSERT(quant_options.has_value());
```

Comment:
Wouldn't it be faster to test if `quant_options.has_value()` rather than having to do three conditions on dtype.